### PR TITLE
Docs: Add GitHub templates

### LINKED ISSUE #663: Issue / PR templates for GH
With the introduction of the GHPages and the development team in need of features and improvements it'd be nice to standardise our issues and pull requests with a common template.

- [ ] Update contributing.md

- [ ] Add issue template

- [ ] Add PR template

Requested by @mdotwills

### PR DESCRIPTION
**Pull request by omgaz**
## Description

Add an issue and pull-request template.

A modification on [CYOA Templates](https://www.talater.com/open-source-templates/#/).

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Motivation and Background Context

Issues and pull-requests require more background context. With multiple projects and developers consuming the library we need a more consistent way of reporting and triaging issues and being able to have context when reviewing pull-requests.

Resolves #663 

## How Has This Been Tested?

Code reviews & this PR.

## Check-list:

- [x] I have read the [Contributing](CONTRIBUTING.md) document.
- [x] I've thought about and labelled my PR/commit message appropriately.
- [x] CI is green (coverage, linting, tests).
- [x] I have updated the documentation accordingly.
- [x] I've two LGTMs/Approvals.
- [x] I've fixed or replied to all my code-review comments.
- [x] I've manually tested with a buddy.
- [x] I've squashed my commits into one.

### CODE REVIEW

File: ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<!--- Provide a general summary of the issue in the Title above -->
+
+## Types of issue
+<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
+- [ ] Bug
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Something else (idea, build tool, documentation improvement)
+
+## Expected Behaviour and Accepted Behaviour
```
Comment (vinteo) [inline ISSUE_TEMPLATE.md]: I feel like there should be a description of some type before this

File: PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<!--- Provide a general summary of your changes in the Title above -->
+
+## Description
+<!--- Describe your changes in detail -->
+
+## Motivation and Context
```
Comment (mdotwills) [inline PULL_REQUEST_TEMPLATE.md]: `Background context` perhaps as per other repos?

File: PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+<!--- If it fixes an open issue, please link to the issue here. -->
+
+## How Has This Been Tested?
+<!--- Please describe in detail how you tested your changes. -->
+<!--- Include details of your testing environment, and the tests you ran to -->
+<!--- see how your change affects other areas of the code, etc. -->
+
+## Screenshots (if appropriate):
+
+## Types of changes
```
Comment (lightbringer1991) [inline PULL_REQUEST_TEMPLATE.md]: is this a bit duplicated with the commit convention we're using?
I mean looking at the PR description/commit message should be enough already.

Comment (sonhanguyen) [reply PULL_REQUEST_TEMPLATE.md]: think so too but having it here doesn't hurt I guess

**Comment by omgaz** [comment]
@lightbringer1991 updated with:

```diff
+## Does this PR introduce a breaking change?
+<!--- If this PR contains a breaking change, please describe the impact and migration path for existing applications. -->
+
+- [ ] Yes
+- [ ] No
```

**Comment by lightbringer1991** [comment]
merging...

**Pull request merged**